src/jdk.jdeps/share/classes/com/sun/tools/javap/BasicWriter.java

```diff
@@ -46,6 +72,20 @@ protected BasicWriter(Context context) {
             throw new AssertionError();
     }
 
+    protected Set<AccessFlag> flagsReportUnknown(AccessFlags flags) {
+        return maskToAccessFlagsReportUnknown(flags.flagsMask(), flags.location());
+    }
+
+    protected Set<AccessFlag> maskToAccessFlagsReportUnknown(int mask, AccessFlag.Location location) {
+        try {
+            return AccessFlag.maskToAccessFlags(mask, location);
+        } catch (IllegalArgumentException ex) {
+            mask &= LOCATION_MASKS.get(location);
+            report(ex);
```

Comment:
Unfortunately the original exception message is missing any info that it is related to access flags and it is not very clear what "Error: Unmatched bit position 0x2 for location CLASS" means.
I would add at least a message prefix, for example "Error: Access Flags: Unmatched bit position 0x2 for location CLASS".

Reply:
Agreed, even though this message is already printed right by the access modifiers.